src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/import.py

```diff
@@ -0,0 +1,90 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.util import CLIError
+from azure.mgmt.containerregistry.v2018_02_01_preview.models import (
+    ImportImageParameters,
+    ImportSource,
+    ImportMode
+)
+from ._utils import (
+    validate_managed_registry,
+    get_registry_by_login_server,
+    get_registry_by_registry_name
+)
+
+SOURCE_REGISTRY_MISING = "Please specify the source container registry:"
+IMPORT_NOT_SUPPORTED = "Image imports are only supported for managed registries."
+INVALID_SOURCE_IMAGE = "Please specify source image in the form of 'registry.azurecr.io/repository[:tag]'."
+SOURCE_REGISTRY_NOT_FOUND = "Source registry cannot be found in the current subscription. " \
+                            "Please specify the full resource ID for it: "
+NO_TTY_ERROR = "Unable to prompt for resource ID as no tty available."
+REGISTRY_MISMATCH = "Registry mismatch. Please check either source-image or resource ID " \
+                    "to make sure that they are referring to the same registry and try again."
+
+
+def acr_import(cmd,
+               client,
+               registry_name,
+               source,
+               resource_id=None,
+               target_tags=None,
+               resource_group_name=None,
+               repository=None,
+               force=False):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, IMPORT_NOT_SUPPORTED)
+
+    slash = source.find('/')
+
+    if slash < 0:
+        if not resource_id:
+            from knack.prompting import prompt, NoTTYException
+            try:
+                resource_id = prompt(SOURCE_REGISTRY_MISING)
+            except NoTTYException:
+                raise CLIError(NO_TTY_ERROR)
+        registry_by_name = get_registry_by_registry_name(cmd.cli_ctx, resource_id)
+        registry_by_login_server = get_registry_by_login_server(cmd.cli_ctx, resource_id)
```

Comment:
Calling list operation twice will be very expensive. Consider combining them if you need to query both name and login server.